pkg/agent/controller/networkpolicy/reconciler.go

```diff
@@ -62,13 +52,24 @@ type Reconciler interface {
 // servicesHash is used to uniquely identify Services.
 type servicesHash string
 
-// hashServices uses the spew library which follows pointers and prints
-// actual values of the nested objects to ensure the hash does not change when
-// a pointer changes.
+// hashServices calculates the hash value of the provided services based on its
+// numbered ports.
+// It constructs the string with strings.Builder instead of using the spew
+// library for efficiency consideration as this function is called quite often.
+// It ignores the difference of protocol and non resolved ports because the hash
+// value is only used to distinguish resolving results for the same services.
 func hashServices(services []v1beta1.Service) servicesHash {
```

Comment:
is the name "hash" still accurate, it doesn't seem that we are doing any hashing operation any more

Reply:
"servicesKey" works for me, but also "servicesIdentifier", based on that comment:
```
// servicesHash is used to uniquely identify Services.
type servicesHash string
```

Reply:
and i like "normalizeServices"

Reply:
we have used normalizeXYZ in the past.. +1

Reply:
Done